contracts/contracts/strategies/NativeStaking/ValidatorRegistrator.sol

```diff
@@ -54,8 +54,8 @@ abstract contract ValidatorRegistrator is Governable, Pausable {
         EXIT_COMPLETE // validator has funds withdrawn to the EigenPod and is removed from the SSV
     }
 
-    event RegistratorChanged(address newAddress);
-    event StakingMonitorChanged(address newAddress);
+    event RegistratorChanged(address indexed newAddress);
+    event StakingMonitorChanged(address indexed newAddress);
     event ETHStaked(bytes pubkey, uint256 amount, bytes withdrawal_credentials);
```

Comment:
Should we also index all of the event parameters that contain the `bytes pubkey`? Might be useful to be able to quickly find the relevant events of a particular validator being staked. 

The documentation states: 
```
When you attempt to index dynamic data types like `string`, `bytes`, `array`, or `struct` in Solidity, they don't get stored in their original form. Instead, the Ethereum log system stores the Keccak-256 hash of these data types.
```

I haven't tested it, but seems we would just need to keccak256(pubkey) in order to query the events of a particular `pubkey`

Reply:
I tried but Waffle doesn't support it :-(

Reply:
Oh I though it just fails with tests. Fails on compile / execution? Ah that's too bad.

Reply:
compile and execution works. The Waffle `emit.withArgs` matcher doesn't work

Reply:
If I understand correctly, we can index the contract event, just not use the Waffle convenience functions to verify the correct behaviour. Possible approaches: 
1. we could introduce another redundant `bytes32` parameter called `pubkeyHash` which contains the keccak256 of the pubkey and index that. 
2. write an auxiliary test suite function to test for the correct value of indexed parameters. Unfortunately the test function would probably need to be aware that the `bytes` property is indexed.

I think it could be really useful to be able to quickly fetch all of the stake related events even before we have the monitoring backend setup and running.

Reply:
> When you attempt to index dynamic data types like string, bytes, array, or struct in Solidity, they don't get stored in their original form. Instead, the Ethereum log system stores the Keccak-256 hash of these data types.

from https://coinsbench.com/understanding-indexed-strings-in-solidity-events-19ba75986de6

I've gone with option 1 and added an indexed `pubKeyHash` to the events